src/Sentry/IHub.cs

```diff
@@ -69,5 +69,17 @@ public interface IHub :
         /// Ends the currently active session.
         /// </summary>
         void EndSession(SessionEndStatus status = SessionEndStatus.Exited);
+
+
+        /// <summary>
+        /// Captures an event with a configurable scope.
+        /// </summary>
+        /// <remarks>
+        /// This allows modifying a scope without affecting other events.
+        /// </remarks>
+        /// <param name="evt">The event to be captured.</param>
+        /// <param name="configureScope">The callback to configure the scope.</param>
+        /// <returns></returns>
+        public SentryId CaptureEvent(SentryEvent evt, Action<Scope> configureScope);
```

Comment:
I Would rename configureScope by withScope or simply scope. Mostly because ConfigureScope gives the impression that it'll behave the same as SentrySDK.Configurescope, thus, not passing an isolated scope to the user.